neo4j/session_with_context.go

```
@@ -784,12 +859,24 @@ func (s *sessionWithContext) resolveHomeDatabase(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-	s.log.Debugf(log.Session, s.logId, "Resolved home database, uses db '%s'", defaultDb)
-	s.config.DatabaseName = defaultDb
-	s.resolveHomeDb = false
+	s.pinHomeDatabase(ctx, defaultDb)
 	return nil
 }
 
+func (s *sessionWithContext) pinHomeDatabase(ctx context.Context, database string) {
+	if !s.resolveHomeDb {
+		return
+	}
+
+	key, _ := computeCacheKey(ctx, s.auth, s.cache, s.config.ImpersonatedUser)
```

:eyes: I spy with my little eye a swallowed error. Is this on purpose? Do we want to continue with a key that might be `""`?

---

Yeah looks like this happens if we fail to get the auth info from the manager, and at that point, we probs don't want to use the empty key for the cache. Added a check and logging for when it errors out.